models/app.model.js

```diff
@@ -22,20 +22,42 @@ exports.fetchArticle = (id) => {
     });
 };
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `
+exports.fetchArticles = (sort_by = "created_at", order = "DESC") => {
+  const validSortBy = [
+    "created_at",
+    "article_id",
+    "title",
+    "topic",
+    "author",
+    "votes",
+    "comment_count",
+  ];
+  if (!validSortBy.includes(sort_by.toLowerCase())) {
+    return Promise.reject({ status: 404, msg: "column not found" });
+  } else if (sort_by !== "comment_count") {
+    sort_by = "a." + sort_by;
+  }
+
+  const validOrder = ["ASC", "DESC"];
+  if (!validOrder.includes(order.toUpperCase())) {
+    return Promise.reject({ status: 400, msg: "bad request" });
```

Comment:
Nice logic, but to make the if statements more readable you could also alternatively save these conditions to a variable as well